Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/StoreOperation/ADLSGen2FSOperation.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ * <p/>
+ * All rights reserved.
+ * <p/>
+ * MIT License
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.sdk.storage.adlsgen2;
+
+import com.microsoft.azure.hdinsight.sdk.common.SharedKeyHttpObservable;
+import com.microsoft.azure.hdinsight.spark.common.ADLSGen2Deploy;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.methods.HttpPatch;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.BufferedHttpEntity;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.InputStreamEntity;
+import rx.Observable;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+
+public class ADLSGen2FSOperation {
+    private SharedKeyHttpObservable http;
+
+    @NotNull
+    private List<NameValuePair> createDirReqParams;
+
+    @NotNull
+    private List<NameValuePair> createFileReqParams;
+
+    @NotNull
+    private List<NameValuePair> appendReqParams;
+
+    @NotNull
+    private ADLSGen2ParamsBuilder flushReqParamsBuilder;
+
+    public ADLSGen2FSOperation(SharedKeyHttpObservable http) {
```

Comment:
We may need to consider use base HttpObservable as parameter type here:
```suggestion
    public ADLSGen2FSOperation(HttpObservable http) {
```